File: dotnet/Multiplayer/Rooms/Updates/PollUpdatesRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Updates
{
    /// <summary>
    /// Request data for polling updates in a room.
    /// Serializes filter parameters for API transmission.
    /// </summary>
    internal class PollUpdatesRequest
    {
        [JsonInclude]
        private string From_players { get; set; } = ERoomTargetPlayers.Host.ToString().ToLower();
        [JsonInclude]
        private int[]? From_players_ids { get; set; }
        [JsonInclude]
        private string? Last_update { get; set; }

        /// <summary>
        /// Initializes a new PollUpdatesRequest.
        /// </summary>
        /// <param name="fromPlayers">Which players to receive updates from.</param>
        /// <param name="fromPlayersIds">Specific player IDs to receive updates from.</param>
        /// <param name="lastUpdate">Only receive updates after this update ID.</param>
        public PollUpdatesRequest(ERoomTargetPlayers fromPlayers = ERoomTargetPlayers.Host, int[]? fromPlayersIds = null, string? lastUpdate = null)
        {
            From_players = fromPlayers.ToString().ToLower();
            From_players_ids = fromPlayersIds;
            Last_update = lastUpdate;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/updates/PollUpdatesRequest.java
package com.michitai.multiplayer.rooms.updates;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.rooms.actions.ERoomTargetPlayers;

/**
 * Request for polling updates from specific players.
 */
public class PollUpdatesRequest {
    @JsonProperty("from_players")
    private String fromPlayers;

    @JsonProperty("from_players_ids")
    private int[] fromPlayersIds;

    @JsonProperty("last_update")
    private String lastUpdate;

    public PollUpdatesRequest(ERoomTargetPlayers fromPlayers, int[] fromPlayersIds, String lastUpdate) {
        this.fromPlayers = fromPlayers.name().toLowerCase();
        this.fromPlayersIds = fromPlayersIds;
        this.lastUpdate = lastUpdate;
    }

    public String getFromPlayers() {
        return fromPlayers;
    }

    public void setFromPlayers(String fromPlayers) {
        this.fromPlayers = fromPlayers;
    }

    public int[] getFromPlayersIds() {
        return fromPlayersIds;
    }

    public void setFromPlayersIds(int[] fromPlayersIds) {
        this.fromPlayersIds = fromPlayersIds;
    }

    public String getLastUpdate() {
        return lastUpdate;
    }

    public void setLastUpdate(String lastUpdate) {
        this.lastUpdate = lastUpdate;
    }
}
